sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2164,10 +2164,20 @@ class Analyzer(
     }
 
     private def hasNestedGenerator(expr: NamedExpression): Boolean = {
+      def hasUnsupportedInnerGenerator(g: Generator): Boolean = g match {
```

Comment:
what does "unsupported" mean here? Looking at the code it just matches `Generator`.

Reply:
my bad... How about just `hasInnerGenerator `?